### PR TITLE
Create new Class that creates a captions srt file,

### LINKED ISSUE #1: Convert Video to Text
Use [Google Cloud Speech-to-Text API](https://cloud.google.com/speech-to-text/docs/async-time-offsets#speech_transcribe_async_word_time_offsets_gcs-protocol) to convert Video to Text file. 

- [x] Read the [Google Cloud Speech-to-Text API](https://cloud.google.com/speech-to-text/docs/async-time-offsets#speech_transcribe_async_word_time_offsets_gcs-protocol) documentation.
- [x] Import video from CLI #4
- [x] Test that Text block has been created

### PR DESCRIPTION
**Pull request by Miguel-Enrique13**
# Description

Created new Captions class. Which creates an .srt and .txt file of a YouTube videos captions. 

## Related Issues

Closes issue #1 

## Type of Change

- [ ] Bug Fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist

- [x] The pull request is for branch `develop`
- [x] `black` gave 0 errors
- [x] `flake` gave 0 errors
- [x] Squashed commits into a single one

# Screenshots of working

Insert screenshots of working samples of the change/feature made.

### CODE REVIEW

**Review by m0mosenpai — APPROVED** [review]

File: script.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import re
+import pysrt
+import logging
+from pathlib import Path
+from pytube import YouTube
+
+logger = logging.getLogger(__name__)
+
+
+class Captions:
```
Comment (KimaruThagna) [inline script.py:11]: Add doc strings as best practice

File: script.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+            os.rename(f"captions ({en_key}).srt", f"{self.video_path}/captions.srt")
+            return f"{self.video_path}/captions.srt"
+
+        except Exception as e:
+            logger.error(e)
+            logger.error("clipbit: error in downloading captions.")
+
+    def generate_script(self):
+        srt_file = pysrt.open(self.srt_filename)
```
Comment (KimaruThagna) [inline script.py:54]: Add function docstrings

**Review by KimaruThagna — COMMENTED** [review]
Code seems good from a pseudo-code point of view. Needs comments and docstrings nonetheless.
Why are you saving the contents in a file? Why not parse everything and have a main file that just calls functions with data returned from other functions?

File: script.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        self.video_path = os.getcwd() + "/videos/{}".format(
+            self.yt.title.lower().replace(" ", "_")
+        )
+        Path(self.video_path).mkdir(parents=True, exist_ok=True)
+        self.srt_filename = self.create_srt()
+        self.generate_script()
+
+    def create_srt(self):
+        try:
+            caption_dict = self.yt.captions.lang_code_index
```
Comment (gmelodie) [inline script.py]: Couldn't you finish your `try ... except` block after this line? I'm not 100% sure but I think it's here where the captions "download" is made (or checked). This avoids the rest of the code to be indented

File: script.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+            en_key = None
+            for key in caption_dict.keys():
+                if re.findall("^en.*", key):
+                    en_key = key
+                    break
```
Comment (gmelodie) [inline script.py]: I'd briefly comment out what is happening here and why this is needed.

File: script.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+            if en_key:
+                self.yt.captions[en_key].download("captions")
+            elif caption_dict.get("a.en", None):
```
Comment (gmelodie) [inline script.py:39]: Maybe this:

```suggestion
            elif "a.en" in caption_dict:
```

**Review by gmelodie — COMMENTED** [review]
Looking good, just a few remarks.

File: script.py (hunk 5 above)
Comment (Miguel-Enrique13) [reply script.py:39]: Would not work, we need None to be returned in case there isn't a "a.en" key

**Pull request merged**